Core/GDCore/IDE/Events/ExpressionCompletionFinder.h

```diff
@@ -80,13 +80,15 @@ struct ExpressionCompletionDescription {
    */
   static ExpressionCompletionDescription ForText(
       const gd::String& type_,
+      const gd::ParameterMetadata *parameterMetadata_,
```

Comment:
Uber nitpicking: using a pointer to store an "optional reference to something, that you're not owning" is fine, but prefer to keep using references in the interface (i.e: in ForText and in the SetParameterMetadata/GetParameterMetadata).
In other words, you can pass a const `gd::ParameterMetadata & parameterMetadata_` (and store its pointer `parameterMetadata = &parameterMetadata_;`).
With a pointer, there is a possible misunderstanding about if the class is taking ownership of the pointer or not.

(You can create a "HasParameterMetadata" if you need to know if this was set or not).